knimin/static/js/platemapper.js

```diff
@@ -0,0 +1,506 @@
+
+  var layout = this._construct_layout();
+  $.post($(location).attr('href'), {action: 'save', layout: JSON.stringify(layout), plate_id: this.plateId}, function(data) {
+    console.log(data);
+  });
+}
+
+/**
+ *
+ * Proceeds to the plate extraction page
+ *
+ **/
+PlateMap.prototype.extractPlate = function () {
+  console.log('extractPlate');
+  var layout = this._construct_layout();
+  var form = $('<form>', {action: $(location).attr('href'), method: 'post'});
+  var fields = {action: 'extract', plate_id: this.plateId,
+                layout: JSON.stringify(layout)};
+  $.each(fields, function(key, val) {
+    $('<input>').attr({type: "hidden", name: key, value: val}).appendTo(form);
+  });
+  form.appendTo('body').submit();
+}
+
+/**
+ *
+ * Keypress event handler on the well input
+ *
+ * @param {Elemnt} current The <input> element where the event has been trigered
+ * @param {Event} e The event object
+ *
+ **/
+PlateMap.prototype.keypressWell = function (current, e) {
+  var row, col;
+  if (e.which === 13) {
+    // The user hit enter, which means that we have to move down one row
+    // Retrieve which is the current row and column
+    row = parseInt($(current).attr('pm-well-row'));
+    col = parseInt($(current).attr('pm-well-column'));
+    // Update indices
+    row = row + 1;
+    if (row === this.rows) {
+      row = 0;
+      col = col + 1;
+      if (col === this.cols) {
+        col = 0;
+      }
+    }
+    // Set the focus to the next input tag
+    this.wellInformation[row][col].inputTag.focus();
+  }
+}
+
+/**
+ *
+ * Change event handler on the well input
+ *
+ * @param {Elemnt} current The <input> element where the event has been trigered
+ * @param {Event} e The event object
+ *
+ **/
+PlateMap.prototype.changeWell = function (current, e) {
+  var sample, sampleInfo, row, col;
+  sample = $(current).val().trim();
+
+  sampleInfo = this.samples[sample];
+
+  // Clean any extra style that we added
+  $(current).removeClass('pm_sample_plated');
+
+  if (sampleInfo === undefined) {
+    // This sample is not recognized - mark the well as problematic
+    $(current).css({'background-color': 'red'});
+  } else {
+    // This sample belongs to a study - label the backgroun with the
+    // same color as the study
+    $(current).css({'background-color': sampleInfo.color});
+    // Check if it has been plated before
+    if (sampleInfo.plates.length > 0) {
+      $(current).addClass('pm_sample_plated');
+    }
+  }
+
+  this.updateWellCommentsArea();
+  // TODO: control the proceed to extraction button
+}
+
+/**
+ *
+ * Show event handler on the well comment modal
+ *
+ **/
+PlateMap.prototype.commentModalShow = function () {
+  var row = parseInt($('#comment-modal-btn').attr('pm-row'));
+  var col = parseInt($('#comment-modal-btn').attr('pm-col'));
+  // Magic number + 1 -> correct index because JavaScript arrays start at 0
+  var wellId = String.fromCharCode('A'.charCodeAt() + row) + (col + 1);
```

Comment:
For some reason I don't foresee the wet lab plating those 1536-well plates manually in the near future (note that this visualization is only used for assisting the wet lab to actually plate the samples).

Can you guys add an issue if you feel strong about it? I don't think code should be blocked due to this.

Regarding padding 0, those well ids are only shown in the interface, they're not stored like that in the DB, so I don't see how that can help downstream.